base/collection_n1ql.go

```diff
@@ -144,13 +144,21 @@ func (c *Collection) BuildDeferredIndexes(indexSet []string) error {
 	return BuildDeferredIndexes(c, indexSet)
 }
 
-func (b *GocbV2Bucket) runQuery(statement string, n1qlOptions *gocb.QueryOptions) (*gocb.QueryResult, error) {
+func (b *GocbV2Bucket) runQuery(scopeName string, statement string, n1qlOptions *gocb.QueryOptions) (*gocb.QueryResult, error) {
 	b.waitForAvailQueryOp()
 
 	if n1qlOptions == nil {
 		n1qlOptions = &gocb.QueryOptions{}
 	}
-	queryResults, err := b.cluster.Query(statement, n1qlOptions)
+
+	var queryResults *gocb.QueryResult
+	var err error
+	if b.IsSupported(sgbucket.BucketStoreFeatureCollections) {
+		scope := b.cluster.Bucket(b.BucketName()).Scope(scopeName)
```

Comment:
I think we already have a handle to a bucket here (b.bucket) that we could be using instead of having the SDK build one via cluster.Bucket().

Reply:
Good call.